build/cmake/MongoSettings.cmake

```
@@ -0,0 +1,299 @@
+        cmake_parse_arguments(validate "" "CODE" "" ${setting_VALIDATE})
+        if(DEFINED validate_CODE)
+            _mongo_eval_cmake("" "${validate_CODE}")
+        endif()
+        if(validate_UNPARSED_ARGUMENTS)
+            message(FATAL_ERROR "Unrecognized VALIDATE options: ${validate_UNPARSED_ARGUMENTS}")
+        endif()
+    endif()
+endfunction()
+
+#[[ Implements DEFAULT setting value logic ]]
+function(_mongo_compute_default outvar arglist)
+    list(APPEND CMAKE_MESSAGE_CONTEXT default)
+    # Clear the value in the caller:
+    unset("${outvar}" PARENT_SCOPE)
+
+    # Parse arguments:
+    cmake_parse_arguments(dflt "" "" "DEVEL;AUDIT" ${arglist})
+
+    # Developer-mode options:
+    if(DEFINED dflt_DEVEL AND MONGODB_DEVELOPER)
+        list(APPEND CMAKE_MESSAGE_CONTEXT "devel")
+        _mongo_compute_default(tmp "${dflt_DEVEL}")
+        message(DEBUG "Detected MONGODB_DEVELOPER: Default of ${setting_NAME} is “${tmp}”")
+        set("${outvar}" "${tmp}" PARENT_SCOPE)
+        return()
+    # Audit-mode options:
+    elseif(DEFINED dflt_AUDIT AND (MONGODB_DEVELOPER OR MONGODB_BUILD_AUDIT))
+        list(APPEND CMAKE_MESSAGE_CONTEXT "audit")
+        _mongo_compute_default(tmp "${dflt_AUDIT}")
+        message(DEBUG "Detected MONGODB_BUILD_AUDIT: Default of ${setting_NAME} is “${tmp}”")
+        set("${outvar}" "${tmp}" PARENT_SCOPE)
+        return()
+    endif()
+
+    # Parse everything else:
+    set(other_args "${dflt_UNPARSED_ARGUMENTS}")
+    cmake_parse_arguments(dflt "" "VALUE;EVAL" "" ${other_args})
+
+    if(DEFINED dflt_VALUE)
+        # Simple value for the default
+        if(DEFINED dflt_EVAL)
+            message(FATAL_ERROR "Only one of VALUE or EVAL may be specified for a DEFAULT")
+        endif()
+        set("${outvar}" "${dflt_VALUE}" PARENT_SCOPE)
+    elseif(DEFINED dflt_EVAL)
+        # Evaluate some code to determine the default
+        _mongo_eval_cmake(DEFAULT "${dflt_EVAL}")
+        set("${outvar}" "${DEFAULT}" PARENT_SCOPE)
+        if(DEFINED DEFAULT)
+            message(DEBUG "Computed default ${setting_NAME} value to be “${DEFAULT}”")
+        else()
+            message(DEBUG "No default for ${setting_NAME} was computed. Default will be an empty string.")
+        endif()
+    elseif(dflt_UNPARSED_ARGUMENTS)
+        message(FATAL_ERROR
+                "${setting_NAME}: "
+                "DEFAULT got unexpected arguments: ${dflt_UNPARSED_ARGUMENTS}")
+    endif()
+endfunction()
+
+#[==[
+Define a new boolean build setting::
+
+    mongo_bool_setting(
+        <name> <doc>
+        [DEFAULT [[DEVEL|AUDIT] [VALUE <value> | EVAL <code>]] ...]
+        [VALIDATE [CODE <code>]]
+        [ADVANCED] [ALLOW_OTHER_VALUES]
+    )
+
+This is a shorthand for defining a boolean setting. See mongo_setting() for more
+option information. The TYPE of the setting will be BOOL, and the implicit
+default value for the setting will be ON if neither DEFAULT nor DEFAULT_EVAL are
+specified.
```

```suggestion
default value for the setting will be ON if DEFAULT is not specified.
```

Appears to be no `DEFAULT_EVAL` option.